File: sample/kotlinapp/src/main/java/com/marcinmoskala/kotlinapp/BaseActivity.java
package com.marcinmoskala.kotlinapp;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import activitystarter.ActivityStarter;

public class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivityStarter.fill(this);
    }
}

File: sample/kotlinapp/src/main/java/com/marcinmoskala/kotlinapp/BaseActivity.kt
package com.marcinmoskala.kotlinapp

import activitystarter.ActivityStarter
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

open class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        ActivityStarter.fill(this, savedInstanceState)
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        ActivityStarter.save(this, outState)
    }
}
